### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by avindra**
Hi Dan, 

First thanks for the library :+1:  I started writing my own version of this and stopped when I found yours. 

I created some tests (this includes travis integration... I think you'll you have to enable it on your side from your account).

I plan to write some more tests so that we can have them, and for my own understanding of the API you've created.

---

I have two questions:
- Would you be opposed to using PSR-2 for coding standards?
- Would you be ok with dropping PHP 5.3 support? Maybe even 5.4 as well? Both are EOL.

### CODE REVIEW

**Comment by dcarbone** [comment]
Hey, sorry for the delay in getting back to you!

Firstly, thanks for your work on this!  I'm not really opposed to psr-2, but what would be the reason to drop php 5.3 and 5.4 support? More precisely, what changes would you like to make that would cause it to no longer run under those versions?

**Comment by dcarbone** [comment]
Looking at the code the only 5.3 breaking change I can see is the use of `[]` vs `array()`, which I personally do not feel is a good enough reason to drop support for something many people still use.

**Comment by avindra** [comment]
Hi Dan- I mean to drop support because even the PHP core team does not
support 5.3 (or 5.4) anymore. They will have permanent security
vulnerabilities and should not be used anywhere by anyone actively
maintaining code.

I'm of the opinion "move fast," where people using legacy software should
stick with legacy versions of everything and not expect others in the
ecosystem to support them. I think supporting EOL software can cripple a
project in the long run, because you always have to be aware of what weird
edge cases you might break. You are ultimately limited to the feature set
of them lowest version you are supporting (5.3 is pretty bad imo).

But as this is your library, you make the call. If you want to support 5.3,
I will update the PR so you can merge it with tests passing on 5.3. Let me
know.

**Comment by dcarbone** [comment]
After thinking about it I honestly don't care enough to take a stand on it, haha. I appreciate the work you're doing and as a general rule I agree with your upgrading statements. My only hesitation is that this lib just does relatively simple string and file manipulation, which has no inherent requirement that precludes it from running under a 5.3 or 5.4 interpreter. 

If you would prefer to update the `composer.json` file to stipulate 5.4+ to support `JsonSerializable` interface functionality and the updated array syntax, I'd be ok with that. If your work requires something that is just not available in 5.4 we can address it then. 

tl;dr: I'm ok with 5.4+ and will merge your changes if you update the `composer.json` file to reflect the requirement.  Sound good?

**Comment by avindra** [comment]
Updated PR. Tests passing:

https://travis-ci.org/avindra/tns-parser/builds/132286593

**Comment by dcarbone** [comment]
Looks good, thanks!
